stdlib/Dates/src/types.jl

```diff
@@ -197,6 +197,8 @@ function validargs(::Type{DateTime}, y::Int64, m::Int64, d::Int64,
     return argerror()
 end
 
+DateTime(dt::Base.Libc.TmStruct) = Time(dt.hour, dt.min, dt.sec)
```

Comment:
Let's do this:
  * take a look at the `Libc.TmStruct` definition [here](https://github.com/JuliaLang/julia/blob/c5be8145fbd9b1f7acd68e81ca23a6a4976c46a9/base/libc.jl#L154)
  * for the `DateTime(dt::Base.Libc.TmStruct)` definition, let's make sure we also pull the year, month, day out of the `TmStruct` as well
  * add a separate `Time(dt::Base.Libc.TmStruct)` constructor that does what you have here

Sound good?

Reply:
Part of the reasoning for my comment is to ensure that the `DateTime` constructor actually returns a `DateTime`. Having a separate `Time` constructor that returns a `Time` and does what you have here is also appropriate, or I guess people could just do `Time(DateTime(tmstruct))`

Reply:
@quinnj I have also added matching `Date` constructor, too!